### DISCUSSION TITLE
What is the suggested way instead of custom config?

### DISCUSSION THREAD

**Discussion by yadavan88** [sbt/sbt#7229] [Q&A]
Hi,
I used the SBT custom config to separate a few types of tests. For example, I had created a custom config as `win` where there were some specific tests related to windows OS and kept them in `src/win/scala`. These tests are not generally ran on CI and was explicitly ran using `win:test`. 
But, in this [link](https://eed3si9n.com/sbt-drop-custom-config/)  there is a mention on custom config being discouraged and going to be dropped. 
So what is the alternative for this? In this case, I only want to separate the tests from the `test` dir and not executed unless explicitly ran. 

I recently wrote a blog about some of the SBT features, there also I mentioned about this usecase. Here is the link for ref: 
https://yadukrishnan.live/unlocking-the-power-of-sbt-a-beginners-guide-to-understanding-not-so-common-features

**Comment by adpi2** [comment]
The recommended way is to create a new project that would only contain the Windows tests:
```scala
lazy val win = project.in(file(win))
  .dependsOn(myProject)
  .libraryDependencies(
    "org.scalatest" %% "scalatest-flatspec" % "3.2.15" % Test
  ) 
```

You would move the Windows tests to `win/src/test` and then you can run them with `win / test`.

**Reply by yadavan88** [reply]
Thanks for the quick response. I understood. But I found the separation within the module more intuitive. Also, doesn't this add too many sub modules to the project? In my case, the repo itself was like a mono repo with more than 10 sub modules. So if I have 2 specific cases (win and some other iot config), I would have 30 sub modules. That was the basis of my confusion. 

Does the same apply to integration tests as well? I used `src/it/scala` in multiple projects..
